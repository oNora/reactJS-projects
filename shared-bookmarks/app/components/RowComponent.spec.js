jest.unmock('./RowComponent');

import React from 'react';
import { shallow, mount } from 'enzyme';

import RowComponent from './RowComponent';

function setup() {
    const props = {
        bookmark: {}
    };

    return shallow(<RowComponent {...props}/>);
}

describe( 'RowComponent page tests', () => {
    const wrapper = setup();

    it('renders correctly', () => {
         expect(wrapper).toMatchSnapshot();
    });

    it('wrapper type should be tr', () => {
        expect(wrapper.type()).toBe('tr');
    });

    it('button elements should be 2', () => {
        expect( wrapper.find( "button" ).length ).toEqual( 2 );
    });
});