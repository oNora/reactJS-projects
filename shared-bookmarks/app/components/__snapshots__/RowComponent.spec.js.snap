// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowComponent page tests renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <tr>
    <td>
        <a
            href={undefined}
        />
    </td>
    <td />
    <td>
        <button
            className="btn btn-primary"
            onClick={undefined}
        >
            <span
                className="glyphicon glyphicon-pencil"
            />
            <span
                className="hidden-xs"
            >
                 Edit 
            </span>
        </button>
        <button
            className="btn btn-danger"
            onClick={undefined}
        >
            <span
                className="glyphicon glyphicon-remove"
            />
            <span
                className="hidden-xs"
            >
                 Delete 
            </span>
        </button>
    </td>
</tr>,
  "nodes": Array [
    <tr>
      <td>
            <a
                  href={undefined}
            />
      </td>
      <td />
      <td>
            <button
                  className="btn btn-primary"
                  onClick={undefined}
            >
                  <span
                        className="glyphicon glyphicon-pencil"
                  />
                  <span
                        className="hidden-xs"
                  >
                         Edit 
                  </span>
            </button>
            <button
                  className="btn btn-danger"
                  onClick={undefined}
            >
                  <span
                        className="glyphicon glyphicon-remove"
                  />
                  <span
                        className="hidden-xs"
                  >
                         Delete 
                  </span>
            </button>
      </td>
</tr>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RowComponent
        bookmark={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RowComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bookmark": Object {},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <tr>
          <td>
                    <a
                              href={undefined}
                    />
          </td>
          <td />
          <td>
                    <button
                              className="btn btn-primary"
                              onClick={undefined}
                    >
                              <span
                                        className="glyphicon glyphicon-pencil"
                              />
                              <span
                                        className="hidden-xs"
                              >
                                         Edit 
                              </span>
                    </button>
                    <button
                              className="btn btn-danger"
                              onClick={undefined}
                    >
                              <span
                                        className="glyphicon glyphicon-remove"
                              />
                              <span
                                        className="hidden-xs"
                              >
                                         Delete 
                              </span>
                    </button>
          </td>
</tr>,
        "_debugID": 2,
        "_renderedOutput": <tr>
          <td>
                    <a
                              href={undefined}
                    />
          </td>
          <td />
          <td>
                    <button
                              className="btn btn-primary"
                              onClick={undefined}
                    >
                              <span
                                        className="glyphicon glyphicon-pencil"
                              />
                              <span
                                        className="hidden-xs"
                              >
                                         Edit 
                              </span>
                    </button>
                    <button
                              className="btn btn-danger"
                              onClick={undefined}
                    >
                              <span
                                        className="glyphicon glyphicon-remove"
                              />
                              <span
                                        className="hidden-xs"
                              >
                                         Delete 
                              </span>
                    </button>
          </td>
</tr>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RowComponent
    bookmark={Object {}}
/>,
}
`;
